### PR TITLE
CMCL-826: Paste VirtualCamera component onto prefab works for pipeline subcomponents

### PR DESCRIPTION
**Pull request by glabute**
### Purpose of this PR

Fix for https://jira.unity3d.com/browse/CMCL-829. 

All cases are now correctly handled for VirtualCamera and FreeLook, except when the CinemachineVirtualCamera or FreeLook component values are pasted onto an object that is in the assets folder without entering Prefab mode.  In this case, the subcomponent values are not pasted.  This case cannot be handled easily, and I suggest that we just leave it as is, since there is an easy workaround: enter prefab edit mode.

### Testing status

- [ ] Added an automated test
- [ ] Passed all automated tests
- [x] Manually tested 

### Documentation status

- [x] Updated [CHANGELOG](https://keepachangelog.com/en/1.0.0/)
- [ ] Updated README (if applicable)
- [ ] Commented all public classes, properties, and methods
- [ ] Updated user documentation

### Technical risk

Lowish, but all combinations should be carefully tested

### Comments to reviewers

All combinations need to be tested.  All should succeed or fail gracefully.
- FreeLook/CinemachineVirtualCamera
- Copy component from sceneobject/prefabInAssetFolder/prefabEditMode
- Paste as new component into sceneobject/prefabInAssetFolder/prefabEditMode
- Paste component values into sceneobject/prefabInAssetFolder/prefabEditMode
- Add VirtualCamera/FreeLook component to empty gameobject for sceneobject/prefabInAssetFolder/prefabEditMode
- Undo and redo of all combinations and permutations

### CODE REVIEW

**Review by gaborkb — APPROVED** [review]
Nice! I manually tested it on my side too.

**Review by gaborkb — CHANGES_REQUESTED** [review]
All combinations need to be tested.  All should succeed or fail gracefully.
- [x] FreeLook/CinemachineVirtualCamera
- [x] Copy component from sceneobject/prefabInAssetFolder/prefabEditMode
- [x] Paste as new component into sceneobject/prefabInAssetFolder/prefabEditMode. **NOTE**: cannot add to prefabInAssetFolder as new, but I think it is gracefully handled (error message with explanation on Freelook in prefab).
- [x] Paste component values into sceneobject/prefabInAssetFolder/sceneobject. **NOTE**: prefabInAssetFolder does not get the values for the rigs, but sceneobject and sceneobject do.
- [x] Add VirtualCamera/FreeLook component to empty gameobject for sceneobject/prefabInAssetFolder/prefabEditMode. **NOTE**: prefabInAssetFolder handled gracefully.
- [ ] Undo and redo of all combinations and permutations. **NOTE**: Not sure if tested all, but could not undo changes in prefabs.

Additional notes:
Reset crashed my unity on a prefab freelook to which I pasted. 
Reproduce:
1. Create a freelook
2. Make it into prefab
3. Copy some freelook values, and paste it to the prefabInAssetFolder.
4. Click on Reset on prefabInAssetFolder.

**Comment by glabute** [comment]
I cannot repro the crash

**Comment by gaborkb** [comment]
> I cannot repro the crash

I cannot reproduce it in 2022.2.0a13 or 2021.3.0f1.
But I can reproduce it in a clean project - unity version 2020.3.34f1.

**Comment by glabute** [comment]
Still cannot repro

File: com.unity.cinemachine/Editor/Editors/CinemachineFreeLookEditor.cs [hunk 1]
```diff
@@ -60,6 +60,17 @@ protected override void OnDisable()
...
         public override void OnInspectorGUI()
         {
+            Target.GetRig(0); // force an update of the rig cache
+            if (!Target.RigsAreCreated)
+            {
+                EditorGUILayout.HelpBox(
+                    "It's not possible to add this component to a prefab instance.  "
+                    + "Instead, you can open the Prefab in Prefab Mode or unpack "
+                    + "the Prefab instance to remove the Prefab connection.",
+                    MessageType.Error);
```
Comment (AntoineCharton) [inline com.unity.cinemachine/Editor/Editors/CinemachineFreeLookEditor.cs:70]: Make sure Jessica reviewed the text.

**Review by AntoineCharton — APPROVED** [review]
I didn't experience a crash (Tested on Mac OS). Worked well on my side. 

The test failing on Ubuntu HDRP is probably infrastructure related. Do your best to make it pass but I think it's an upgrade issue of HDRP that messes up with graphic code (Occlusion code for example). 

Check this test to add it to the ignore list if needed.
Tests.Runtime.CamerasBlendingTests.BlendingBetweenCameras

**Pull request merged**